### ISSUE TITLE
Plotting labels to runs\detect\train2\labels.jpg...  module 'backend_interagg' has no attribute 'FigureCanvas'

### ISSUE THREAD

**Issue by Bevis-nb** [ultralytics/ultralytics#17995]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

#When I run the "train_programme",I see the feedback:
Plotting labels to runs\detect\train2\labels.jpg...  module 
'backend_interagg' has no attribute 'FigureCanvas'

#In addition, after calling the trained best. pt to recognize images, the feedback is' no detection '

### Additional

python = 3.10.11
ultralytics = 8.3.38

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Bevis-nb, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) for helpful resources where you can find detailed examples for [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, as well as answers to many common questions.

If this is a 🐛 **Bug Report**, it would be very helpful to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) so we can better understand and debug the issue.

Given the feedback you’ve shared, it seems there could be an issue with a backend visualization tool causing the `'backend_interagg' has no attribute 'FigureCanvas'` error. To help us investigate further, please share detailed steps to reproduce the issue, including:

1. The script or commands you're running
2. Any custom settings or modifications (if applicable)

If this is related to detection issues (like the `'no detection'` feedback you’ve mentioned), please also share a snippet of your training logs, example images, or specific parameters used for training or inference. Additionally, ensure you're following the [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) for optimal performance.

---

Join the Ultralytics community where it suits you best:
- For real-time assistance, head over to [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth discussions? Check out [Discourse](https://community.ultralytics.com) 🛠️
- Share insights and learnings on our [Subreddit](https://reddit.com/r/Ultralytics) ⏫

---

## Upgrade

Please ensure you're using the latest `ultralytics` package with all dependencies installed. Run the following command to upgrade:

```bash
pip install -U ultralytics
```

This ensures the latest features and bug fixes are in effect.

---

## Environments

Consider running YOLO in one of our supported environments for optimized performance and minimal setup issues:

- **Free GPU Notebooks**:  
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a>  
  <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a>  
  <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud Platforms**: Discover our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) and [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: Use our pre-built [Docker Image](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

---

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>  
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing, validating correct operation across all [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/).

---

This is an automated response to help address your query faster. An Ultralytics engineer will review your issue and provide further assistance as soon as possible 😊

**Comment by Y-T-G** [comment]
Can you post the full traceback?

Can you provide the output after running this command in terminal: `yolo checks`?

**Comment by Bevis-nb** [comment]
### run"yolo checks"
Ultralytics 8.3.41 🚀 Python-3.10.11 torch-2.5.1+cpu CPU (Intel Xeon E5-2430 0 2.20GHz)
Setup complete ✅ (12 CPUs, 15.9 GB RAM, 72.5/133.8 GB disk)

OS                  Windows-10-10.0.19045-SP0
Environment         Windows
Python              3.10.11
Install             pip
RAM                 15.92 GB
Disk                72.5/133.8 GB
CPU                 Intel Xeon E5-2430 0 2.20GHz
CPU count           12
GPU                 None
GPU count           None
CUDA                None

numpy               ✅ 2.1.3>=1.23.0
numpy               ✅ 2.1.3<2.0.0; sys_platform == "darwin"
matplotlib          ✅ 3.9.3>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 11.0.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.5.1>=1.8.0
torch               ✅ 2.5.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.20.1>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 6.1.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.12>=2.0.0

### I use labelme to annotate pictures, and then use labelme2yolo --json_dir command to convert json files into yolo format.
### train.py
from ultralytics import YOLO
model = YOLO('yolov8n.pt')
model.train(data="train/json/YOLODataset/dataset.yaml", epochs=2, imgsz=640, batch=8)  # train the model
### show:
Ultralytics 8.3.41 🚀 Python-3.10.11 torch-2.5.1+cpu CPU (Intel Xeon E5-2430 0 2.20GHz)
engine\trainer: task=detect, mode=train, model=yolov8n.pt, data=train/json/YOLODataset/dataset.yaml, epochs=2, time=None, patience=100, batch=8, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=None, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs\detect\train
Overriding model.yaml nc=80 with nc=2

                   from  n    params  module                                       arguments                     
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]                 
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]                
  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]             
  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]             
  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           
  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              
  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]           
  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]                 
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]                 
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]                  
 16                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]                
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 18                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]                 
 19                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 21                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]                 
 22        [15, 18, 21]  1    751702  ultralytics.nn.modules.head.Detect           [2, [64, 128, 256]]           
Model summary: 225 layers, 3,011,238 parameters, 3,011,222 gradients, 8.2 GFLOPs

Transferred 319/355 items from pretrained weights
Freezing layer 'model.22.dfl.conv.weight'
train: Scanning \\?\G:\Projects\CS2yolo\train\json\YOLODataset\labels\train... 76 images, 0 backgrounds, 0 corrupt: 100%|██████████| 76/76 [00:02<00:00, 29.25it/s]
train: New cache created: \\?\G:\Projects\CS2yolo\train\json\YOLODataset\labels\train.cache
val: Scanning \\?\G:\Projects\CS2yolo\train\json\YOLODataset\labels\val... 19 images, 0 backgrounds, 0 corrupt: 100%|██████████| 19/19 [00:00<00:00, 29.58it/s]
val: New cache created: \\?\G:\Projects\CS2yolo\train\json\YOLODataset\labels\val.cache
Plotting labels to runs\detect\train\labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.001667, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 0 dataloader workers
Logging results to runs\detect\train
Starting training for 2 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        1/2         0G      1.365      3.704      1.243         10        640: 100%|██████████| 10/10 [01:29<00:00,  8.93s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:08<00:00,  4.37s/it]
                   all         19         22    0.00369      0.955      0.471      0.313
  0%|          | 0/10 [00:00<?, ?it/s]
      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        2/2         0G       1.28       3.06      1.114          5        640: 100%|██████████| 10/10 [01:24<00:00,  8.47s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:08<00:00,  4.43s/it]
                   all         19         22    0.00368      0.955      0.563      0.384

2 epochs completed in 0.054 hours.
Optimizer stripped from runs\detect\train\weights\last.pt, 6.2MB
Optimizer stripped from runs\detect\train\weights\best.pt, 6.2MB

Validating runs\detect\train\weights\best.pt...
Ultralytics 8.3.41 🚀 Python-3.10.11 torch-2.5.1+cpu CPU (Intel Xeon E5-2430 0 2.20GHz)
Model summary (fused): 168 layers, 3,006,038 parameters, 0 gradients, 8.1 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:07<00:00,  3.81s/it]
                   all         19         22    0.00368      0.955      0.563      0.384
                    ct         10         11     0.0038          1      0.548      0.384
                     t         10         11    0.00356      0.909      0.579      0.383
Speed: 9.6ms preprocess, 301.0ms inference, 0.0ms loss, 59.1ms postprocess per image
Results saved to runs\detect\train

**Comment by Bevis-nb** [comment]
### trained result
![confusion_matrix](https://github.com/user-attachments/assets/ab3e50af-a652-49d9-8a93-bce5debd028f)
![confusion_matrix_
![R_curve](https://gith
[results.csv](https://github.com/user-attachments/files/18017118/results.csv)
ub.com/user-attachments/assets/637a5840-a304-457b-8bab-c769ae0db93b)

![labels](https://githu
![PR_curve](https://github.com/user-attachments/assets/0a3b92d7-f369-4a78-a736-4f2cd33a8bcd)

![P_curve](https://github.com/user-attachments/assets/97f72f81-4ec4-4b87-8708-2fefad4a74be)

![labels_correlogram](https://github.com/user-attachments/assets/faf8ee80-ff74-4371-a1d1-1a10bb977f7f)
b.com/user-attachments/assets/5b9b6e0a-226b-421c-8870-14134bb6ade7)
normalized](https://git
![Uploading results.p
![train_batch0](https://github.com/user-attachments/assets/e50c2772-2b6b-45e6-8edd-9507f14ffd68)
ng…]()
![Uploading train_batch1.jp
![train_batch2](h
![Uploading val_batc
![val_batch1_pred](https://github.com/user-attachments/assets/378e1a67-17ed-4764-9119-5667a1e1a600)
h0_pred.
![val_batch1_labels](https://github.com/user-attachments/assets/f5d26518-be61-416b-a980-013a8cbc387e)
jpg…]()
ttps://github.com/user-attachments/assets/7c344c47-bb9c-48bf-949c-768312707f97)
g…]()
![val_batch0_labels](https://github.com/user-attachments/assets/c2c55323-e66a-471e-8e2e-2e743a3dff09)


hub.com/user-attachments/assets/39801d5e-2546-475b-9516-21b1f2f54827)
![F1_curve](https://github.com/user-attachments/assets/2dc61aff-7968-47a4-9157-07bc9dd4b69d)

**Comment by Y-T-G** [comment]
There's no error in your loss. 

You only trained for 2 epochs. The model wouldn't let anything with 2 epochs. And you have very few images. You need a lot more.

https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/

**Comment by Bevis-nb** [comment]
Then I will try to increase the number of epochs.

**Comment by Bevis-nb** [comment]
Then I will try to increase the number of epochs.

**Comment by glenn-jocher** [comment]
Increasing the number of epochs is a good step. Start with at least 300 epochs to establish a solid training baseline. If overfitting occurs, reduce the epochs accordingly. You can refer to [this guide](https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/) for optimization tips.
